File: Recursion/002.Sum of First N Natural Numbers/SumOfFirstN.cpp
#include <iostream>
using namespace std;

int sumOfFirstN(int n) {
    if (n == 1) {
        return 1;
    }
    return n + sumOfFirstN(n-1);
}

int main() {

    int n;
    cout<<"Enter the limit: ";
    cin>>n;
    cout<<"Sum of first "<<n<<" number is: "<< sumOfFirstN(n);
    return 0;
}

File: Recursion/002.Sum of First N Natural Numbers/SumOfFirstN.java
import java.util.Scanner;

class SumOfFirstN {

    static int sumOfFirstN(int n) {
        if (n == 1) {
            return 1;
        }
        return n + sumOfFirstN(n-1);
    }
    public static void main(String args[]) {
        System.out.print("Enter the limit: ");
        Scanner input = new Scanner(System.in);
        int number = input.nextInt();
        System.out.print("Sum of first "+ number + " number is " + sumOfFirstN(number));
        input.close();
    }
}